### PR TITLE
tests: add bmc module test cases

### PR DESCRIPTION
**Pull request by anandaravuri**

### CODE REVIEW

**Comment by ibarraza** [comment]
Please run without the cover.sh script, there are a lot of build warnings that will need to be fixed and the script will not show them due to the script using "CMAKE_BUILD_TYPE=Coverage".

File: testing/xfpga/test_bmc_c.cpp [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+	std::array<fpga_token, 2> tokens_;
+	fpga_handle handle_;
+	fpga_properties filter_;
+	uint32_t num_matches_;
+	test_platform platform_;
+	test_system *system_;
+
+};
+
+fpga_result bmc_c_p ::write_sysfs_file(fpga_token token, const char* file, void *buf, size_t count)
```
Comment (r-rojo) [inline testing/xfpga/test_bmc_c.cpp]: have a space before `::`

File: testing/xfpga/test_bmc_c.cpp [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+	EXPECT_EQ(bmcGetLastPowerdownCause(tokens_[0], &string), FPGA_OK);
+	if (string) {
+		free(string);
+		string = NULL;
+	}
+	EXPECT_NE(bmcGetLastPowerdownCause(tokens_[0],NULL), FPGA_OK);
+
+	// Get firmware version
+	EXPECT_EQ(bmcGetFirmwareVersion(tokens_[0], &version), FPGA_OK);
+	if (string) {
```
[the remark this answers is not in the document]
Comment (anandaravuri) [reply testing/xfpga/test_bmc_c.cpp]: fixed

File: testing/xfpga/test_bmc_c.cpp (hunk 1 above)
Comment (anandaravuri) [reply testing/xfpga/test_bmc_c.cpp]: fixed

**Pull request merged**